### ISSUE TITLE
Latest Watchexec Library won't compile

### ISSUE THREAD

**Issue by calebschoepp** [watchexec/watchexec#522]
I'm trying to use the Watchexec library in a Rust project. I'm on MacOS. I've added the following to my `Cargo.toml`

```toml
watchexec = "2.1.1"
```

When I run `cargo build` I get the errors in the codeblock below. This is the same issue as #506 but I don't know what the workaround is for the library (as opposed to the CLI).

```
error[E0277]: the trait bound `BStr: AsRef<[_; 0]>` is not satisfied
  --> /Users/caleb/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/store/packed/find.rs:95:35
   |
95 |                 .map(|(_rest, r)| r.name.as_bstr().as_ref())
   |                                   ^^^^^^^^^^^^^^^^ ------ required by a bound introduced by this call
   |                                   |
   |                                   the trait `AsRef<[_; 0]>` is not implemented for `BStr`
   |
   = help: the following other types implement trait `AsRef<T>`:
             <BStr as AsRef<BStr>>
             <BStr as AsRef<[u8]>>

error[E0277]: the trait bound `BStr: AsRef<[_; 0]>` is not satisfied
  --> /Users/caleb/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/store/packed/find.rs:91:33
   |
91 |         a.binary_search_by_key(&full_name.as_ref(), |b: &u8| {
   |                                 ^^^^^^^^^ ------ required by a bound introduced by this call
   |                                 |
   |                                 the trait `AsRef<[_; 0]>` is not implemented for `BStr`
   |
   = help: the following other types implement trait `AsRef<T>`:
             <BStr as AsRef<BStr>>
             <BStr as AsRef<[u8]>>

error[E0283]: type annotations needed
    --> /Users/caleb/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/fullname.rs:109:43
     |
109  |             if let Some(shortened) = name.strip_prefix(category.prefix().as_ref()) {
     |                                           ^^^^^^^^^^^^ -------------------------- type must be known at this point
     |                                           |
     |                                           cannot infer type of the type parameter `P` declared on the associated function `strip_prefix`
     |
     = note: cannot satisfy `_: core::slice::SlicePattern`
note: required by a bound in `core::slice::<impl [T]>::strip_prefix`
    --> /Users/caleb/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/slice/mod.rs:2284:28
     |
2284 |     pub fn strip_prefix<P: SlicePattern<Item = T> + ?Sized>(&self, prefix: &P) -> Option<&[T]>
     |                            ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `core::slice::<impl [T]>::strip_prefix`
help: consider specifying the generic argument
     |
109  |             if let Some(shortened) = name.strip_prefix::<P>(category.prefix().as_ref()) {
     |                                                       +++++

error[E0283]: type annotations needed
   --> /Users/caleb/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/fullname.rs:109:43
    |
109 |             if let Some(shortened) = name.strip_prefix(category.prefix().as_ref()) {
    |                                           ^^^^^^^^^^^^                   ------ type must be known at this point
    |                                           |
    |                                           cannot infer type of the type parameter `P` declared on the associated function `strip_prefix`
    |
    = note: multiple `impl`s satisfying `BStr: AsRef<_>` found in the `bstr` crate:
            - impl AsRef<BStr> for BStr;
            - impl AsRef<[u8]> for BStr;
help: consider specifying the generic argument
    |
109 |             if let Some(shortened) = name.strip_prefix::<P>(category.prefix().as_ref()) {
    |                                                       +++++

error[E0283]: type annotations needed
   --> /Users/caleb/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/fullname.rs:132:46
    |
132 |         } else if let Some(shortened) = name.strip_prefix(Category::MainPseudoRef.prefix().as_ref()) {
    |                                              ^^^^^^^^^^^^                                  ------ type must be known at this point
    |                                              |
    |                                              cannot infer type of the type parameter `P` declared on the associated function `strip_prefix`
    |
    = note: multiple `impl`s satisfying `BStr: AsRef<_>` found in the `bstr` crate:
            - impl AsRef<BStr> for BStr;
            - impl AsRef<[u8]> for BStr;
help: consider specifying the generic argument
    |
132 |         } else if let Some(shortened) = name.strip_prefix::<P>(Category::MainPseudoRef.prefix().as_ref()) {
    |                                                          +++++

error[E0283]: type annotations needed
   --> /Users/caleb/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/fullname.rs:139:18
    |
139 |             name.strip_prefix(Category::LinkedPseudoRef { name: "".into() }.prefix().as_ref())
    |                  ^^^^^^^^^^^^                                                        ------ type must be known at this point
    |                  |
    |                  cannot infer type of the type parameter `P` declared on the associated function `strip_prefix`
    |
    = note: multiple `impl`s satisfying `BStr: AsRef<_>` found in the `bstr` crate:
            - impl AsRef<BStr> for BStr;
            - impl AsRef<[u8]> for BStr;
help: consider specifying the generic argument
    |
139 |             name.strip_prefix::<P>(Category::LinkedPseudoRef { name: "".into() }.prefix().as_ref())
    |                              +++++

error[E0283]: type annotations needed
    --> /Users/caleb/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/raw.rs:76:18
     |
76   |                 .strip_prefix(namespace.0.as_bstr().as_ref())
     |                  ^^^^^^^^^^^^ ------------------------------ type must be known at this point
     |                  |
     |                  cannot infer type of the type parameter `P` declared on the associated function `strip_prefix`
     |
     = note: cannot satisfy `_: core::slice::SlicePattern`
note: required by a bound in `core::slice::<impl [T]>::strip_prefix`
    --> /Users/caleb/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/slice/mod.rs:2284:28
     |
2284 |     pub fn strip_prefix<P: SlicePattern<Item = T> + ?Sized>(&self, prefix: &P) -> Option<&[T]>
     |                            ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `core::slice::<impl [T]>::strip_prefix`
help: consider specifying the generic argument
     |
76   |                 .strip_prefix::<P>(namespace.0.as_bstr().as_ref())
     |                              +++++

error[E0283]: type annotations needed
  --> /Users/caleb/.cargo/registry/src/github.com-1ecc6299db9ec823/git-ref-0.22.0/src/raw.rs:76:18
   |
76 |                 .strip_prefix(namespace.0.as_bstr().as_ref())
   |                  ^^^^^^^^^^^^                       ------ type must be known at this point
   |                  |
   |                  cannot infer type of the type parameter `P` declared on the associated function `strip_prefix`
   |
   = note: multiple `impl`s satisfying `BStr: AsRef<_>` found in the `bstr` crate:
           - impl AsRef<BStr> for BStr;
           - impl AsRef<[u8]> for BStr;
help: consider specifying the generic argument
   |
76 |                 .strip_prefix::<P>(namespace.0.as_bstr().as_ref())
   |                              +++++

```

**Comment by passcod** [comment]
You can probably try using the latest git? for now. I'm working on releasing a new version soon

**Comment by calebschoepp** [comment]
That works! Closing the issue.

```toml
watchexec = { git = "https://github.com/watchexec/watchexec.git", rev = "f83790bce3a20f0e993f8f1ff9b652b645ed7668"}
```

**Comment by AnthonyMichaelTDM** [comment]
when can we expect a new release containing the fixes?

**Comment by passcod** [comment]
This weekend probably

**Comment by AnthonyMichaelTDM** [comment]
Glad to hear that, thank you

**Comment by passcod** [comment]
Should be fixed as of 2.2.0

**Comment by calebschoepp** [comment]
I just updated to `2.2.0` and it successfully compiled. Thanks.
